torch/distributed/_fsdp/fully_sharded_data_parallel.py

```diff
@@ -864,6 +868,8 @@ def update_p_data(output_tensor: torch.Tensor) -> None:
             # Trim any padding and reshape to match original size.
             p.data = p.data[: p._orig_size.numel()].view(p._orig_size)  # type: ignore[attr-defined]
 
+        self._lazy_init()
```

Comment:
why is it called here?

Reply:
There was a parameters-uninitialized corner case when testing.  The case seems to be fixed or not exist anymore. Will remove this.